src/main/antlr2/org/codehaus/groovy/antlr/java/java.g

```diff
@@ -402,10 +402,9 @@ wildcardType
 		(("extends" | "super")=> typeArgumentBounds)?
 	;
 
-typeArgumentsDiamond
-{Token first = LT(1);}
+typeArgumentsOrDiamond
     :   LT! GT!
-    {#typeArgumentsDiamond = #(create(TYPE_ARGUMENTS, "TYPE_ARGUMENTS",first,LT(1)), #typeArgumentsDiamond);}
```

Comment:
Could you explain the reason why the line should be removed?

Reply:
Since this concerns diamond operators I could not think of any case where actual type arguments could be extracted.